core/tests/test_utils.py

```diff
@@ -2282,6 +2280,118 @@ def assert_same_list_elements(self, phrases, stdout):
 GenericTestBase = AppEngineTestBase
 
 
+class GenericEmailTestBase(GenericTestBase):
+    """Base class for tests requiring email services."""
+
+    class EmailMessageMock(python_utils.OBJECT):
+        """Mock for core.platform.models email services messages."""
+
+        def __init__(
+                self, sender_email, recipient_email, subject, plaintext_body,
+                html_body, bcc=None, reply_to=None, recipient_variables=None):
+            self.sender = sender_email
+            self.to = recipient_email
+            self.subject = subject
+            self.body = plaintext_body
+            self.html = html_body
+            self.bcc = bcc
+            self.reply_to = reply_to
+            self.recipient_variables = recipient_variables
+
+    emails_dict = {}
+
+    def run(self, result=None):
+        """Adds a context swap on top of the test_utils.run() method so that
+        test classes extending GenericEmailTestBase will automatically have
+        a mailgun api key, mailgun domain name and mocked version of
+        send_email_to_recipients().
+        """
+        with self.swap(feconf, 'MAILGUN_API_KEY', 'key'), (
```

Comment:
If we're doing this automatically then why are these swaps also present in some of the other test files?